data/src/lib.rs

```diff
@@ -0,0 +1,82 @@
+//! The single source of truth for all data paths.
+//!
+//! Run `cargo make gen-assets` before using this!
+
+pub mod bench;
+pub mod example;
+pub mod ltr;
+pub mod qambert;
+pub mod smbert;
+pub mod test;
+
+use std::{
+    collections::HashMap,
+    env::{current_dir, var_os},
+    fs::File,
+    io::{BufReader, Error, ErrorKind, Result},
+    path::{Path, PathBuf},
+};
+
+use serde::Deserialize;
+use serde_json::{from_reader, from_value, Value};
+
+/// Resolves the path to the requested data.
+fn resolve_path(path: &[impl AsRef<Path>]) -> Result<PathBuf> {
+    let manifest = var_os("CARGO_MANIFEST_DIR")
+        .ok_or_else(|| Error::new(ErrorKind::NotFound, "missing CARGO_MANIFEST_DIR"))?;
+    let manifest = PathBuf::from(manifest).canonicalize()?;
```

Comment:
Why not just use `manifest.parent().ok_or_else(...).to_path_buf()`, and make it independent of the current working directory?